src/day22.scala

```diff
@@ -0,0 +1,114 @@
+// using scala 3.0.2
+
+// inspired by solution https://topaz.github.io/paste/#XQAAAQAeCAAAAAAAAAAzHIoib6p4r/McpYgEEgWhHoa5LSRMlQ+wWN2dSgFba+3OQLAQwPs2n+kH2p5lUvR4TJnE77lx5kFb6HW+mKObQM2MOxuv8Z0Ikuk8EytmoUVdCtg9KGX3EgBJtaZ1KUYGSX44w3DeWgEYnt4217TlxVAwn+5nGWeZWu11MdZIUg0S0DptNm4ymgHBUjFFBjfwS9dAE8lGus2C676rW5dFQTWsPSbX0KPErnY+Au6A3b83ioXEigKgv43S/YV5xd5UPogLgH1bRepf9ZNHvkHnlJ3Fd8zU5YiQjfY5gIE5QhyClAOLQAvcaeuqL1Lwi3AUVhbn0S6PoR9fI8CgSKgHxG/5OMMTxp0XmDbJSozSliQrG9gteNy/c9lwoX3ACTodcPOxhvJDpTAw5ZEJ47i/vPvBLkv+6/0vZYIveS86bML4r9niB2s0A5jOO+JzdYtkpZTnbm9eRXGZRjSdtCoXHwprlF308Xe+6+HHDqBhy7cGh5CwT9+SnwIVdPGJYQh3e5VAOI1I5+9bI+B2L91PDrMpszHrymcQHgJTbwLVPMyQ0oC2Gx5/2RDpcGyzxQVmOXvtmF47wD44rTALuAor27Bcc5HoPORpHW3ZJ8O3L/fz30m1bWv8EIYeWY1jdgX0lB98R7+bVHrqnzVsmKVAy+rIXnDqMJvojQF1a3Reetfg83JQTIbJoa+jnghFw/hYZ+thAB54sovYyutIFGGWx5JknARI3wngn+iEmbhxO3lM5Z8PiLES69y6erunAmEzXwlL6hMvTtx1znp3sp8GoYk4AyZJ/sFaukNpX4970vioZf+sZ+7rzJ4bKUiBc1fuebalSH2EJoT9Bkf33IU/OfkgZXgv067jeaY9Ktu+3oxELBs9Ea6g80BsTb3Xe33WaL1DUbpwTOw304VRILT9/dyVGg==
+
+package day22
+
+import scala.util.Using
+import scala.io.Source
+
+import Command.*
+import scala.collection.mutable.ListBuffer
+
+@main def part1(): Unit =
+  println(s"The solution is ${part1(readInput())}")
+
+@main def part2(): Unit =
+  println(s"The solution is ${part2(readInput())}")
+
+def readInput(): String =
+  Using.resource(Source.fromFile("input/day22"))(_.mkString)
+
+case class Dimension(min: Int, max: Int):
+  require(min <= max)
+
+  def isSubset(d: Dimension): Boolean =
+    min >= d.min && max <= d.max
+
+  infix def insersect(d: Dimension): Option[Dimension] =
+    Option.when(max >= d.min && min <= d.max) {
+      (min max d.min) by (max min d.max)
+    }
+
+  def size: Int = max - min + 1
+
+extension (x1: Int)
+  infix def by (x2: Int): Dimension = Dimension(x1, x2)
+
+case class Cuboid(xs: Dimension, ys: Dimension, zs: Dimension):
+  def volume: BigInt = BigInt(xs.size) * ys.size * zs.size
+
+enum Command:
+  case On, Off
+
+case class Step(command: Command, cuboid: Cuboid)
+
+case class Cube(x: Int, y: Int, z: Int)
+
+def intersect(old: Cuboid, curr: Cuboid): Option[Cuboid] =
```

Comment:
To be consistent with `Dimension.intersect`, consider making this an infix method of `Cuboid`.